src/main/kotlin/racingcar/domain/Analyzer.kt

```diff
@@ -0,0 +1,28 @@
+package racingcar.domain
+
+import racingcar.data.Car
+
+class Analyzer {
+    fun judgeWinner(cars: List<Car>): List<Car> {
+        val winners = mutableListOf<Car>()
+        var maxLocation = NOT_YET_SET
+
+        for (car in cars) {
+            if (car.location > maxLocation) {
+                winners.clear()
+                winners.add(car)
+                maxLocation = car.location
+            } else if (car.location == maxLocation) {
+                winners.add(car)
+            }
+        }
+        return winners.toList()
+    }
```

Comment:
[1, 2, 3] 아래와 같이 최대값을 탐색한 뒤, 최대값과 location 이 일치하는 car 객체만 winners에 담도록 filter 함수를 사용하는 방법으로 수정한다면 코드를 간결하게 줄일 수 있을 것 같아요!
```suggestion
   fun judgeWinner(cars: List<Car>): List<Car> {
        val bestScore = cars.maxOf { it.location }
        return cars.filter { it.location == bestScore }
    }
```